x/ecocredit/module/module.go

```diff
@@ -36,43 +39,50 @@ func (a AppModule) DefaultGenesis(codec.JSONMarshaler) json.RawMessage {
 func (a AppModule) ValidateGenesis(codec.JSONMarshaler, sdkclient.TxEncodingConfig, json.RawMessage) error {
 	return nil
 }
-
-func (a AppModule) RegisterRESTRoutes(sdkclient.Context, *mux.Router) {}
-
-func (a AppModule) RegisterGRPCGatewayRoutes(sdkclient.Context, *runtime.ServeMux) {}
-
-func (a AppModule) GetTxCmd() *cobra.Command {
+func (a AppModule) InitGenesis(sdk.Context, codec.JSONMarshaler, json.RawMessage) []abci.ValidatorUpdate {
 	return nil
 }
 
-func (a AppModule) GetQueryCmd() *cobra.Command {
+func (a AppModule) ExportGenesis(sdk.Context, codec.JSONMarshaler) json.RawMessage {
 	return nil
 }
 
-func (a AppModule) InitGenesis(sdk.Context, codec.JSONMarshaler, json.RawMessage) []abci.ValidatorUpdate {
+func (a AppModule) GetTxCmd() *cobra.Command {
 	return nil
 }
 
-func (a AppModule) ExportGenesis(sdk.Context, codec.JSONMarshaler) json.RawMessage {
-	return nil
-}
+func (a AppModule) GetQueryCmd() *cobra.Command { return client.QueryCmd() }
+
+func (a AppModule) RegisterGRPCGatewayRoutes(sdkclient.Context, *runtime.ServeMux) {}
 
 func (a AppModule) RegisterInvariants(sdk.InvariantRegistry) {}
 
-func (a AppModule) Route() sdk.Route { return sdk.Route{} }
+func (a AppModule) RegisterServices(cfg module.Configurator) {
+	ecocredit.RegisterMsgServer(cfg.MsgServer(), a.Srv)
+	ecocredit.RegisterQueryServer(cfg.QueryServer(), a.Srv)
+}
 
-func (a AppModule) QuerierRoute() string { return "" }
+func (a AppModule) BeginBlock(sdk.Context, abci.RequestBeginBlock) {}
 
-func (a AppModule) LegacyQuerierHandler(*codec.LegacyAmino) sdk.Querier {
+func (a AppModule) EndBlock(sdk.Context, abci.RequestEndBlock) []abci.ValidatorUpdate {
 	return nil
 }
 
-func (a AppModule) RegisterServices(configurator module.Configurator) {
-	server.RegisterServices(a.Key, configurator)
+/**** DEPRECATED ****/
+
+// Should we deprecate it? RegisterServices already registers everything for the module.
+func (a AppModule) RegisterInterfaces(codectypes.InterfaceRegistry) {}
```

Comment:
No it's definitely not deprecated. Rather it's actually rather new.

Reply:
Tests are actually failing because the msg service descriptor needs to be wired up there

Reply:
yes, thanks for discussion in Discord. I added the implementation for it and added AppModuleBasic.